Add Job and Bridge Creation Pages

NavyAdmiral opened the pull request.

NavyAdmiral commented:
@rupurt [Wanted to test the container properly without shallow rendering, but I keep getting unknown node with tag 13 error.](https://raw.githubusercontent.com/smartcontractkit/chainlink/2e3dd5ca788be54bdcd0134d5475ea48d77e72e1/gui/__tests__/containers/Create.test.js)

rupurt commented:
@NavyAdmiral I'm not exactly sure why that error is happening but doesn't happen when I remove the `<BridgeForm />` component.

NavyAdmiral commented:
@rupurt Aside from being untested, are there other big issues with this PR?

NavyAdmiral commented:
@rupurt I'm starting to think that the `node with tag 13 error` is because of Formik 😞

NavyAdmiral commented:
@rupurt Alright, it's as I thought. There's a bug with Enzyme and this has been reported both in Formik and Enzyme repos.
[Enzyme](https://github.com/airbnb/enzyme/issues/1509)
[Formik](https://github.com/jaredpalmer/formik/issues/719)
There's also a workaround presented in issue #719. 
`yarn add --dev enzyme-adapter-react-16@npm:enzyme-react-adapter-future`
What are your thoughts about using this new enzyme adapter to test Create container?

NavyAdmiral commented on gui/src/containers/Jobs.js:
```diff
@@ -70,17 +81,22 @@ export class Jobs extends Component {
           </Grid>
-          <Grid item xs={3}>
-            {renderSidebar(this.props)}
+          <Grid item>
```
Right of the grid? If I understand you correctly, I didn't like aligning it that way because Drawer slid over it when opened.

NavyAdmiral commented on gui/src/containers/Create.js:
```diff
@@ -0,0 +1,101 @@
+    // Need to set int value because <Tabs/> component
+    // will not focus on tab with strings
+    if (this.props.match) {
+      switch (this.props.match.params.structure) {
```
@rupurt That will not work if we add Create Service Agreement in this tab container.

NavyAdmiral commented on gui/src/containers/Create.js:
```diff
@@ -0,0 +1,101 @@
+    if (this.props.history) {
+      switch (value) {
+        case 0:
+          this.props.history.replace('/create/bridge')
```
am I wrong to think `.push` should make it work?

NavyAdmiral commented on gui/src/components/BridgeForm.js:
```diff
@@ -0,0 +1,121 @@
+  handleSubmit (values, { props }) {
+    const formattedValues = JSON.parse(JSON.stringify(values).replace('confirmations', 'defaultConfirmations'))
+    formattedValues.defaultConfirmations = parseInt(formattedValues.defaultConfirmations) || 0
+    props.submitCreate('v2/bridge_types', formattedValues, true)
```
@rupurt what I'm doing here is changing `confirmations` from redux store to `defaultConfirmations` for api. thsoe 2 lines can be removed if we change `confirmations` to `defaultConfirmations` in redux, but I thought `confirmations` was a more suitable word for redux.

NavyAdmiral commented:
@rupurt addressed most of the comments; I'm not too happy with create button positionings

NavyAdmiral commented:
@rupurt addressed most of the comments including the create button positioning. I think I liked both create job/create bridge buttons on the `/ || /jobs`` page more though

j16r commented:
Just looking at the form, a couple of thoughts:

  1. When I submitted apparently invalid JSON, the submit button was disabled, but there didn't appear to be a way to enable it again.
  2. The submit button in the form isn't obvious until I move my mouse cursor over it, like the sign in form, the button should have a clear demarcation from its surroundings.
  3.  The JSON field should probably be a text area, rather than an input box.
  4. The verbs around fields are unnecessary "e.g: Type Bridge Name", the noun should be sufficient, e.g: "Paste JSON" should be "Job Spec"
  5. Not sure how I feel about combining all the forms into the one GUI with the tabs to select different functions, could be a bit limiting.

NavyAdmiral commented:
@j16r in what way do you think combining all the forms could be limiting?

j16r commented:
It limits the opportunities to take cues from the context in which it is launched, e.g: imagine that the form for creating a job is at the top of the jobs index. Then you can view example jobs and you can see when the job is added live as you add the job. If you can change the "Create Job" form to another form, then this context no longer makes sense.

NavyAdmiral commented:
@j16r Agreed, plus it made statemanagement somewhat contrived. Thanks for the tips. If there's anything else, let me know

rupurt commented on gui/src/components/JobForm.js:
```diff
@@ -0,0 +1,95 @@
+const styles = theme => ({
+  jsonfield: {
+    paddingTop: theme.spacing.unit * 1.25,
+    width: theme.spacing.unit * 150
```
This should use the grid instead of trying to set a width. On smaller screens this causes the text field to overflow.

<img width="947" alt="screen shot 2018-08-20 at 9 50 28 am" src="https://user-images.githubusercontent.com/680789/44354881-df4cc980-a45f-11e8-86e0-983a406b8063.png">

NavyAdmiral replied:
@rupurt good catch. fixed this and wondering if the same should be done on BridgeForm. (bridgeform inputs have small width by default and I doubt this can ever be a problem there)

rupurt replied:
@NavyAdmiral yeah, I think we should also do it for bridge types. Though I think we should try to make the text fields roughly the same size they are now, even on bigger screens. You can do that by using the `xs` & `xl` props on the `<Grid item>...` component.

I also feel like we should make the create job textarea wider.

NavyAdmiral commented:
Will reopen a new PR as this one got messy.
